### PR TITLE
Map all available midi input messages

### PR DESCRIPTION
**Pull request by flosse**

### CODE REVIEW

File: src/devices/pioneer_ddj_400/mod.rs [hunk 1]
```diff
@@ -20,11 +20,27 @@ pub const MIDI_DEVICE_DESCRIPTOR: &MidiDeviceDescriptor = &MidiDeviceDescriptor
 pub enum Deck {
     /// Left
     One,
+    /// Right
+    Two,
+}
 
+#[derive(Debug, Clone, Copy)]
+pub enum Channel {
```
Comment (uklotzde) [inline src/devices/pioneer_ddj_400/mod.rs]: Why? The controller does not distinguish between *deck* and *channel*.

**Review by uklotzde — CHANGES_REQUESTED** [review]
Mapping to a `ControlIndex` becomes much more complicated with too many and nested enums. Try this first before unnecessarily fleshing out those enums. Neither needed nor helpful, I have made this mistake before.

File: src/devices/pioneer_ddj_400/mod.rs (hunk 1 above)
Comment (flosse) [reply src/devices/pioneer_ddj_400/mod.rs]: :ok_hand:

File: src/input/mod.rs [hunk 2]
```diff
@@ -342,50 +343,35 @@ impl From<SliderEncoderInput> for ControlValue {
...
-}
-
-impl From<SliderInput> for Input {
-    fn from(from: SliderInput) -> Self {
-        Self::Slider(from)
-    }
+/// A switch with multiple positions.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SwitchInput {
+    pub position: u8,
```
Comment (uklotzde) [inline src/input/mod.rs]: Why not use u32 directly and avoid the fallible conversion?

File: src/devices/pioneer_ddj_400/input.rs [hunk 3]
```diff
@@ -1,54 +1,153 @@
 // SPDX-FileCopyrightText: The djio authors
 // SPDX-License-Identifier: MPL-2.0
 
+//! # Pioneer DDJ-400
+//!
+//! Most of the terms in this module have been taken
+//! from the manual (`DDJ-400_manual_Manual_EN.pdf`).
+//! The manual and detailed information can be found here:
+//! <https://support.pioneerdj.com/hc/en-us/sections/4416577146009-ddj-400>.
```
Comment (uklotzde) [inline src/devices/pioneer_ddj_400/input.rs]: https://www.pioneerdj.com/-/media/pioneerdj/software-info/controller/ddj-400/ddj-400_midi_message_list_e1.pdf

Comment (flosse) [reply src/devices/pioneer_ddj_400/input.rs]: Oh no, if i had this a few hours ago, i could have saved a lot of time :disappointed:

**Pull request merged**